File: CSharp/OpenTraceability/Models/MasterData/VocabularyElement.cs
using Newtonsoft.Json.Linq;
using OpenTraceability.Interfaces;
using OpenTraceability.Utility;
using OpenTraceability.Utility.Attributes;
using System;
using System.Collections.Generic;

namespace OpenTraceability.Models.MasterData
{
    public class VocabularyElement : IVocabularyElement
    {
        public string ID { get; set; }

        public string EPCISType { get; set; }

        [OpenTraceabilityJson("@type")]
        public string JsonLDType { get; set; }

        public JToken Context { get; set; }

        public VocabularyType VocabularyType
        {
            get
            {
                VocabularyType type = VocabularyType.Unknown;
                foreach (VocabularyType t in Enum.GetValues(typeof(VocabularyType)))
                {
                    if (EnumUtil.GetEnumDescription(t).Trim().ToLower() == EPCISType?.Trim().ToLower())
                    {
                        type = t;
                    }
                }
                // if we can't determine the type from the EPCIS type, then try to determine it from the JSON-LD type
                if (type == VocabularyType.Unknown)
                {
                    if (!string.IsNullOrEmpty(JsonLDType))
                    {
                        switch(JsonLDType.Trim().ToLower())
                        {
                            case "gs1:seafood":
                            case "gs1:product":
                                type = VocabularyType.Tradeitem;
                                break;
                            case "gs1:organization":
                                type = VocabularyType.TradingParty;
                                break;
                            case "gs1:place":
                                type = VocabularyType.Location;
                                break;
                        }
                    }
                }
                return type;
            }
        }

        public List<IMasterDataKDE> KDEs { get; set; } = new List<IMasterDataKDE>();
    }
}

File: JDK11/opentraceability/src/main/java/opentraceability/models/masterdata/vocabularyelement.java
package opentraceability.models.masterdata;

import java.util.List;
import opentraceability.interfaces.IMasterDataKDE;
import opentraceability.interfaces.IVocabularyElement;
import opentraceability.interfaces.VocabularyType;
import opentraceability.models.identifiers.*;
import opentraceability.utility.EnumUtil;
import org.json.JSONObject;

public class VocabularyElement implements IVocabularyElement {
    public String id = null;
    public String epcisType = "";
    public String jsonLDType = null;
    public JSONObject context = null;

    public VocabularyType vocabularyType = VocabularyType.Unknown;

    public List<IMasterDataKDE> kdes = new List<IMasterDataKDE>();

    public VocabularyElement() {
    }

    public VocabularyType getVocabularyType() {
        VocabularyType type = VocabularyType.Unknown;
        for (VocabularyType t : VocabularyType.values()) {
            if (EnumUtil.GetEnumDescription(t).trim().equals(epcisType.trim())) {
                type = t;
            }
        }
        return type;
    }

    public void setVocabularyType(VocabularyType value) {
        vocabularyType = value;
    }
}